docs/reference/sql/security.asciidoc

```diff
@@ -24,13 +24,43 @@ PKI/X.509:: Use X.509 certificates to authenticate {es-sql} to {es}. For this, o
 [discrete]
 [[sql-security-permissions]]
 ==== Permissions (server-side)
-Lastly, one the server one need to add a few permissions to
-users so they can run SQL. To run SQL a user needs `read` and
+On the server, one needs to add a few permissions to
+users so they can run SQL. To run SQL, a user needs `read` and
 `indices:admin/get` permissions at minimum while some parts of 
-the API require `cluster:monitor/main`. 
+the API require `cluster:monitor/main`.
 
-The following example configures a role that can run SQL in JDBC querying the `test` and `bort`
-indices:
+You can add permissions with either an <<apiexample,API call>>
+or the configuration file <<fileexample,roles.yml>>. File-based role management
+is useful if you want to define a role that doesn't need to change. Also, you
+cannot use the role management APIs to view or edit a role defined in `roles.yml`.
+Using a role management API is the preferred method for defining roles.
```

Comment:
```suggestion
You can add permissions by <<defining-roles,creating a role>>, and assigning
that role to the user. Roles can be created using {kib}, an
<<sql-role-api-example,API call>> or the <<sql-role-file-example,`roles.yml`
configuration file>>. Using {kib} or the role management APIs is the preferred
method for defining roles. File-based role management is useful if you want to
define a role that doesn't need to change. You cannot use the role management
APIs to view or edit a role defined in `roles.yml`. 
```

Reply:
I think it's good to explain that it's a two-step process: 1) defining a role, and 2) assigning that role to the user. It's also good to mention Kibana as a third way of managing roles. And I've reworded a bit, and consistently formatted `roles.yml`.